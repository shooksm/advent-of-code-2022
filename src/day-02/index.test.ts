import {
  puzzleOne,
  puzzleTwo,
  scoringMapPuzzleOne,
  scoringMapPuzzleTwo,
} from './index';
import fs from 'fs';

const exampleData = `A Y
B X
C Z`;

const data = fs.readFileSync('./src/day-02/input.txt', 'utf8');

describe('Day 2', () => {
  describe('scoringMap', () => {
    it('should map out the scoring combinations', () => {
      expect(scoringMapPuzzleOne.size).toBe(9);
      expect(scoringMapPuzzleOne).toMatchSnapshot();
    });
    it('should map out the scoring combinations with additional calculate move by player b', () => {
      expect(scoringMapPuzzleTwo.size).toBe(9);
      expect(scoringMapPuzzleTwo).toMatchSnapshot();
    });
  });

  describe('Puzzle 1', () => {
    it('should return 15 with example input', () => {
      expect(puzzleOne(exampleData)).toBe(15);
    });

    it('should calculate the correct answer', () => {
      expect(puzzleOne(data)).toMatchSnapshot();
    });
  });

  describe('Puzzle 2', () => {
    it('should return 12 with example input', () => {
      expect(puzzleTwo(exampleData)).toBe(12);
    });

    it('should calculate the correct answer', () => {
      expect(puzzleTwo(data)).toMatchSnapshot();
    });
  });
});
