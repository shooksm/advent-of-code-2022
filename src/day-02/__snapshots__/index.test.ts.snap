// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day 2 Puzzle 1 should calculate the correct answer 1`] = `8933`;

exports[`Day 2 Puzzle 2 should calculate the correct answer 1`] = `11998`;

exports[`Day 2 scoringMap should map out the scoring combinations 1`] = `
Map {
  "A X" => 4,
  "A Y" => 8,
  "A Z" => 3,
  "B X" => 1,
  "B Y" => 5,
  "B Z" => 9,
  "C X" => 7,
  "C Y" => 2,
  "C Z" => 6,
}
`;

exports[`Day 2 scoringMap should map out the scoring combinations with additional calculate move by player b 1`] = `
Map {
  "A X" => 3,
  "A Y" => 4,
  "A Z" => 8,
  "B X" => 1,
  "B Y" => 5,
  "B Z" => 9,
  "C X" => 2,
  "C Y" => 6,
  "C Z" => 7,
}
`;
